internal/signing/env_signature_client.go

```diff
@@ -0,0 +1,82 @@
+package signing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+	"github.com/stellar/go/txnbuild"
+)
+
+type envSignatureClient struct {
+	networkPassphrase       string
+	distributionAccountFull *keypair.Full
+}
```

Comment:
Plz, make sure to override the String() method to prevent the secret key from ever being printed. You can do that in another PR.